lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2147,6 +2147,27 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           } else {
             VD->diagnose(diag::property_does_not_override, isClassContext)
                 .highlight(OA->getLocation());
+            //If the override is a property-decl, check if there exists
+            //a no-args func-decl in the base class, if yes, provide a fix-it.
+            auto declName = VD->getBaseName();
+            auto declType = VD->getResultTypeRepr();
+            auto inherited = DC->getSelfClassDecl()->getInherited();
+            //If there is an inherited context, perform member lookup.
+            if (!inherited.empty()) {
+              auto superClass = inherited.front().getType();
+              auto lookupResult = superClass->getAnyNominal()->lookupDirect(declName);
+              
+              for (auto& candidate : lookupResult) {
+                //If the parameter list is empty, compare the return types
+                //of the candidates with the declaration type.
+                if (getParameterList(candidate)->size() == 0) {
```

Comment:
I think a safer option here would be to determine whether there is only ne member with no parameters and a matching result type.

Reply:
You mean to say, we should have an additional check before parameter size, that the lookupResult should have just one candidate? (i.e. `lookupResult.size() == 0` )?

Reply:
You need to determine that there is a single declaration that the new member is trying to override for that you need to make sure that there are no parameters and the result type matches exactly, you can use `llvm::find_first` for that walking over the results.

Reply:
Hmm, I will see that!